test_server.py

```diff
@@ -34,6 +34,25 @@ def test_get_product_list(self):
         self.assertTrue( resp.status_code == HTTP_200_OK )
         self.assertTrue( len(resp.data) > 0 ) 
 
+    def test_get_product_price(self):
+        resp = self.app.get('/products?minPrice=99&maxPrice=999')
+        self.assertTrue( resp.status_code == HTTP_200_OK )
+        data = json.loads(resp.data)
+        self.assertTrue(data[0]['name'] == 'TV')
+        self.assertTrue(data[1]['name'] == 'Blender')
+
+    def test_get_product_limit(self):
```

Comment:
The `test_get_product_price(self)` method is testing the `limit` URL attribute of the GET function, so the 'minPrice' URL attribute is unnecessary.

Moreover, it's not clear whether the desired result of a single element is due to the 'limit' functionality working as desired (what the test is trying to accomplish) or because of the filter on price (e.g. what if the '$99 Blender' test product price changes to $100 at a later time; even if the 'limit' functionality was broken, the test will still return a single value).

As such, IMHO, the `minPrice` attribute should be removed from the request.